src/lib.rs

```diff
@@ -823,9 +848,10 @@ impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterSetBits<'a, T> {
     type Item = usize;
 
     fn next(&mut self) -> Option<usize> {
+        debug_assert!(self.range.end <= self.vob.len);
```

Comment:
Unrelated nitpick: In the line above, shouldn't we have `-> Option<Self::Item>` for maintainability sake?

Reply:
It will never be anything but `usize`: I'm not sure that the extra indirection helps users read the documentation?

Reply:
Up to you. I'd prefer to not duplicate the generic type myself.

Reply:
I'm neutral: you get a static type error if you change one but not the other. But I find rustdoc output harder to read when it says `next(&self) -> Option<Self::Item>` as there's now an extra level of indirection to understand...

Reply:
I don't disagree :)